### ISSUE TITLE
after upgrade v3.2.4 -> 3.3.8, smarty error: "PHP Notice:  Undefined index: percent in .../templates_c/...cb_0.file.list-virtual_mailbox.tpl.php"

### ISSUE THREAD

**Issue by pgnd** [postfixadmin/postfixadmin#470]
After upgrading from Postfixadmin 3.2.4 -> 3.3.8, running
```
php -v
	PHP 7.4.16 (cli) (built: Mar  2 2021 10:35:17) ( NTS )
	Copyright (c) The PHP Group
	Zend Engine v3.4.0, Copyright (c) Zend Technologies
		with Zend OPcache v7.4.16, Copyright (c), by Zend Technologies
```

and running `setup.php`, site appears (so far) to function, but php-fpm logs return lots of these
```
==> /var/log/nginx/php-fpm.log <==
[14-Mar-2021 10:52:07 America/New_York] PHP Notice:  Undefined index: percent in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 114
[14-Mar-2021 10:52:07 America/New_York] PHP Notice:  Trying to access array offset on value of type null in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 114
[14-Mar-2021 10:52:07 America/New_York] PHP Notice:  Undefined index: percent in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 116
```

checking the referenced template
```
	...
	$_smarty_tpl->smarty->ext->_foreach->restore($_smarty_tpl, 1);?>
					</td>
				<?php }?>
				<td><?php echo $_smarty_tpl->tpl_vars['item']->value['name'];?>
	</td>
				<?php if ($_smarty_tpl->tpl_vars['CONF']->value['quota'] === 'YES') {?>
					<td>
					<?php if ($_smarty_tpl->tpl_vars['item']->value['quota'] == 0) {?>
						<?php echo $_smarty_tpl->tpl_vars['PALANG']->value['pOverview_unlimited'];?>

					<?php } elseif ($_smarty_tpl->tpl_vars['item']->value['quota'] < 0) {?>
						<?php echo $_smarty_tpl->tpl_vars['PALANG']->value['pOverview_disabled'];?>

					<?php } else { ?>
						<?php if ($_smarty_tpl->tpl_vars['boolconf_used_quotas']->value) {?>


114							<?php if ($_smarty_tpl->tpl_vars['divide_quota']->value['percent'][$_smarty_tpl->tpl_vars['i']->value] > $_smarty_tpl->tpl_vars['CONF']->value['quota_level_high_pct']) {?>
								<?php $_smarty_tpl->_assignInScope('quota_level', "high");?>
116							<?php } elseif ($_smarty_tpl->tpl_vars['divide_quota']->value['percent'][$_smarty_tpl->tpl_vars['i']->value] > $_smarty_tpl->tpl_vars['CONF']->value['quota_level_med_pct']) {?>
								<?php $_smarty_tpl->_assignInScope('quota_level', "mid");?>
							<?php } else { ?>
								<?php $_smarty_tpl->_assignInScope('quota_level', "low");?>
							<?php }?>
```

**Comment by kao-ua** [comment]
maybe you need to run setup.php to upgrade your db version ?

**Comment by pgnd** [comment]
as mentioned above,

> and running setup.php

already done

**Comment by DavidGoodwin** [comment]
Try changing public/list-virtual.php at around line 324 to have : 

```php
$divide_quota = array('current' => [], 'quota' => [], 'percent' => [], 'quota_width' => []);
```

instead of : 

```php
$divide_quota = array('current' => array(), 'quota' => array());
```

**Comment by pgnd** [comment]
@DavidGoodwin 

with
```
edit public/list-virtual.php
-	$divide_quota = array('current' => array(), 'quota' => array());
+	$divide_quota = array('current' => [], 'quota' => [], 'percent' => [], 'quota_width' => []);
```

now, in logs,
```
==> /var/log/nginx/php-fpm.log <==
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 0 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 114
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 0 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 116
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 1 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 114
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 1 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 116
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 2 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 114
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 2 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 116
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 3 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 114
[15-Mar-2021 06:29:08 America/New_York] PHP Notice:  Undefined offset: 3 in /usr/local/postfixadmin/templates_c/33dfa64a8a86b289113746f5f67930526cd888cb_0.file.list-virtual_mailbox.tpl.php on line 116
```

**Comment by DavidGoodwin** [comment]
I think there's something funky going on with your quota setup, but I don't know what .... as it's not a feature I use, and it seems poorly documented.

In list-virtual.php try changing the  for ($i = 0; $i < sizeof($tMailbox); $i++) { .... } loop at around line 324 to have these defaults in it - 

```PHP
for ($i = 0; $i < sizeof($tMailbox); $i++) {
   $gen_show_status_mailbox [$i] = gen_show_status($tMailbox[$i]['username']);

    // new 4 lines here :
    $divide_quota['current'][$i] = Config::Lang('unknown');
    $divide_quota['quota_width'][$i] = 0;
    $divide_quota['percent'][$i] = null;
    $divide_quota['quota'][$i] = Config::Lang('unknown');
  

    if (isset($tMailbox[$i]['current'])) {
      // as before
   }
}
```

**Comment by pgnd** [comment]
>  something funky going on with your quota setup

so far, I've not _noticed_ any (other) errors or functional issues.

> poorly documented.

not terribly uncommon, right? sigh

> try changing  ...

with
```
edit public/list-virtual.php
	$gen_show_status_mailbox = array();
-	$divide_quota = array('current' => array(), 'quota' => array());
+	$divide_quota = array('current' => [], 'quota' => [], 'percent' => [], 'quota_width' => []);

for ($i = 0; $i < sizeof($tMailbox); $i++) {
	$gen_show_status_mailbox [$i] = gen_show_status($tMailbox[$i]['username']);

+	// new 4 lines here :
+	$divide_quota['current'][$i] = Config::Lang('unknown');
+	$divide_quota['quota_width'][$i] = 0;
+	$divide_quota['percent'][$i] = null;
+	$divide_quota['quota'][$i] = Config::Lang('unknown');

	if (isset($tMailbox[$i]['current'])) {
		$divide_quota ['current'][$i] = divide_quota($tMailbox[$i]['current']);
	}
```

now, in logs, no more `Undefined offset` notices/errors

seems to do the trick!  is that a reasonable _fix_, or a temp workaround?

**Comment by DavidGoodwin** [comment]
I'll commit that fix shortly and merge it into the 3.3 branch.
